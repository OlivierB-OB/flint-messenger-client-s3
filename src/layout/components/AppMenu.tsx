import AppBar from '@material-ui/core/AppBar';
import Grid from '@material-ui/core/Grid';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Forum from '@material-ui/icons/Forum';
import React, { Fragment } from 'react';
import { MyConversations } from '../../conversations/components/MyConverstions';
import { MyConversationsButton } from '../../conversations/components/MyConversationsButton';
import { MyName } from '../../identity/components/MyName';
import { MyProfileLink } from '../../profileForm/components/MyProfileLink';
import { MyContacts } from '../../users/components/MyContacts';
import { MyContactsButton } from '../../users/components/MyContactsButton';

export function AppMenu() {
  return (
    <Fragment>
      <AppBar position="static" style={{ height: '10vh' }}>
        <Grid container justify="space-between" alignItems="center" style={{ height: '100%' }}>
          <Grid item>
            <Toolbar>
              <Forum fontSize="large" />
              <Typography variant="h3">flint.</Typography>
            </Toolbar>
          </Grid>
          <Grid item>
            <Toolbar>
              <MyName />
            </Toolbar>
          </Grid>
          <Grid item>
            <Toolbar>
              <MyConversationsButton />
              <MyContactsButton />
              <MyProfileLink />
            </Toolbar>
          </Grid>
        </Grid>
      </AppBar>
      <MyContacts />
      <MyConversations />
    </Fragment>
  );
}
