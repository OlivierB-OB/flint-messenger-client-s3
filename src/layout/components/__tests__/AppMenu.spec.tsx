import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { AppMenu } from '../AppMenu';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('AppMenu', () => {
  let Connected: any = AppMenu;

  it('should provide allowNavigation to Menu', async () => {
    const store = mockStore({
      layout: {
        allowNavigation: 'allowNavigation from state',
      },
    });
    const component = shallow(<Connected store={store} />).dive();
    expect(component).toMatchSnapshot();
  });
});
