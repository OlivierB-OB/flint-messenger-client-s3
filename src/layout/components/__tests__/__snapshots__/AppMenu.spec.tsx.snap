// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu should provide allowNavigation to Menu 1`] = `
<Menu
  allowNavigation="allowNavigation from state"
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
